core/regexp.rbs

```diff
@@ -1558,11 +1550,63 @@ class Regexp
   #
   # If any regexp pattern contains captures, the behavior is unspecified.
   #
-  def self.union: () -> Regexp
-                | (String | Regexp pat1, *String | Regexp pat2) -> Regexp
-                | (::Array[String | Regexp]) -> Regexp
+  def self.union: (*_ToRegexp | string patterns) -> Regexp
+                | (array[_ToRegexp | string] patterns) -> Regexp
+                | (Symbol | [Symbol] symbol_pattern) -> Regexp
```

Comment:
Technically this `[Symbol]` should be `array[Symbol]` where the return value is an array of length 1, only containing a symbol, but I think that's overkill here.